src/components/app/App.js

```diff
@@ -32,6 +36,9 @@ function App() {
         <Route exact path="/voting-centers">
           <CenterContainer />
         </Route>
+        <Route path="/voting-centers/:id">
+          {state.selectedCenter ? <CenterDetail/> : <Redirect to="/" />}
+        </Route>
```

Comment:
If there is no selected center, it redirects to the home page

Reply:
useReducer + useContext .... tasty! Nice job!